RockLib.Configuration.Remote/IHttpClientFactory.cs

```diff
@@ -0,0 +1,15 @@
+using System.Net.Http;
+
+namespace RockLib.Configuration.Remote;
+
+/// <summary>
+/// A factory to create HttpClient instances.
+/// </summary>
+public interface IHttpClientFactory
```

Comment:
There is an `IHttpClientFactory` [in .NET](https://learn.microsoft.com/en-us/dotnet/api/system.net.http.ihttpclientfactory), so this would be confusing to have another interface with the same name. Is there something the current interface doesn't have that we need?

Reply:
That has `CreateClient(string)`, which takes a configured client name. It didn't seem like a good fit, as the implementation I'm using here (which uses a provided `HttpMessageHandler` factory). There's also an extension method for `CreateClient()`, so that would get pretty muddy.

I think it would be a good idea to give it a different name, though.